crates/sui-prover/src/prove.rs

```diff
@@ -80,6 +90,10 @@ pub struct GeneralConfig {
     /// Boogie running mode
     #[clap(name = "boogie-file-mode", long, short = 'm', global = true,  default_value_t = BoogieFileMode::Function)]
     pub boogie_file_mode: BoogieFileMode,
+
+    /// Lean running mode
+    #[clap(name = "lean-backend", long, short = 'l', global = true)]
```

Comment:
let's make an `enum`, and let's call this option `--backend`, with boogie the default